homework-submission/react-hyf/react-hyf-client/src/components/TodoInput.js

```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react';
+import './todoInput.css';
+
+export default class TodoInput extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: '' };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.addTodo = this.addTodo.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ value: event.target.value });
+  }
+
+  addTodo(todo) {
+    if (todo.length > 0) {
+      this.props.addTodo(todo);
+      this.setState({ value: '' });
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        <input type="text" value={this.state.value} onChange={this.handleChange} />
```

Comment:
You have here a `div` element containing the elements below it, and you have as a parent a `ul` if we have an error Boundary surround the application this will cause an error and the application may never run.

you can find the valid children to `ul` under the example as a permitted content label:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul